## store infromation needed for all protocols
locations:
  kitchen:
    x: 0.5
    y: 1.44
    yaw: 0.78     # 45 degrees ≈ 0.78 rad
  living_room:
    x: 1.94
    y: -0.4
    yaw: -0.3
  home:
    x: 0.5
    y: -1.4
    yaw: 3.14     # π rad (180 degrees)

## need to be integerated to have robot interface get its info 
person_location_state: "person_location"
robot_location_state: "robot_location"

# available type: ## todo set it in a way to verifiy htat yaml doesnt include 
## undefined types: text, audio, video, question_answer 

###
# day:[] or not there means everyday
# optional
# reset_pattern optional         reset_pattern:
          # hours: 1
          # minutes: 30
          # atleast one hours or minutes need to be there

# different types of pattern:
# everytime the protocol is triggered it will be reset
## make sure that /state are turned to false
## reset after 5 mins
# reset_pattern:
#   type: on_trigger 

# reset_pattern:
#   type: periodic
#   hours: 1
#   minutes: 30

## this will be reset depending on the value of blackboard key
# reset_pattern:
#   blackboard_key: last_reset_timestamp 

###
protocols:
  TwoReminderProtocol:
    medicine_am:
      permissible_locations: TO BE INCORPORATED
      low_level:
        first_reminder: "please take your night medicine"
        type_first: text
        second_reminder: "This is the second reminder to take your night medicine"
        type_second: text
        wait_time_between_reminders: 5
      high_level:
        priority: 2
        success_on: {'state': 'coffee', 'value': True}
        requirements: 
          time: {'from': '1:30', 'to': '5:00'}

  XReminderProtocol:
    medicine_am_2:
      low_level:
        reminder_1: "/home/olagh48652/smart-home/src/smart-home-robot/shr_resources/resources/food_reminder.mp3"
        type_1: audio
        wait_1: 10
        reminder_2:  "please take your night medicine"
        type_2: text
        number_of_protocols: 2
      high_level:
        priority: 2
        reset_pattern:
          type: periodic
          minutes: 60
        requirements:
          time: {from: "10:00", to: "11:00"}
          permissible_locations: [living_room]

    medicine_pm_2:
      low_level:
        reminder_1: "/home/olagh48652/smart-home/src/smart-home-robot/shr_resources/resources/food_reminder.mp3"
        type_1: audio
        wait_1: 10
        reminder_2:  "please take your night medicine"
        type_2: text
        wait_2: 10 
        reminder_3: "please 1 take your night medicine"
        type_3: text
        number_of_protocols: 3
      high_level:
        priority: 2
        reset_pattern:
          type: periodic
          minutes: 60
        requirements:
          time: {from: "10:00", to: "11:00"}
          permissible_locations: [living_room]